internal/lifecycle/propagation.go

```diff
@@ -75,25 +74,27 @@ func (p *Propagator) ApplyLabels() client.Object {
 	return p.dst
 }
 
-// extractDeclaredLabelKeys - retrieves the list of label keys from the propagate-labels annotation
+// labelKeysToPropagate - retrieves the list of label keys from the propagate-labels annotation
 // in the source object. Returns nil if missing, invalid, or empty.
-func (p *Propagator) extractDeclaredLabelKeys() []string {
-	annotationVal := strings.TrimSpace(p.src.GetAnnotations()[PropagateLabelsAnnotation])
-	if annotationVal == "" {
+func (p *Propagator) labelKeysToPropagate() []string {
+	var keys []string
+	annotation := strings.TrimSpace(p.src.GetAnnotations()[PropagateLabelsAnnotation])
+	if strings.TrimSpace(annotation) == "" {
```

Comment:
If the annotation is an empty string the previous call to strings.TrimSpace should have cleared any whitespaces. The second call should have no effect, I think
```suggestion
	annotation := strings.TrimSpace(p.src.GetAnnotations()[PropagateLabelsAnnotation])
	if annotation == "" {
```